R/ui.R

```diff
@@ -83,8 +81,47 @@ include_scripts <- function() {
     "js/reset-p-id-and-refresh-browser.js",
     "js/show-footer.js",
     "js/trigger-button.js",
-    "js/navigate-away.js"
-  )
-  lapply(scripts, function(x) shiny::includeScript(system.file(
+    "js/navigate-away.js")
+
+  wrapped_scripts <- lapply(scripts, function(x) shiny::includeScript(system.file(
     x, package = "psychTestR")))
+
+  if(length(opt$additional_scripts) > 0) {
+
+    wrapped_additional_scripts <- lapply(opt$additional_scripts, function(x) {
+      if(base::startsWith(x, "http")) {
+        htmltools::tags$script(src = x)
+      } else {
+        htmltools::includeScript(x)
+      }
+    })
+
+    wrapped_scripts <- c(wrapped_scripts, wrapped_additional_scripts)
+  }
+
+  wrapped_scripts
 }
+
+
+insert_logo <- function(opt, side) {
+
+  if (!is.null(opt$logo)) {
+    logo <- shiny::img(
+      id = "logo",
+      src = opt$logo,
+      style = sprintf("text-align: left; display: inline; width: %s; height: %s;",
+                      width = opt$logo_width,
+                      height = opt$logo_height))
+  } else {
+    logo <- " "
+  }
+
+  if(opt$logo_right & side == "right") {
+    logo
+  } else if(opt$logo_right == FALSE & side == "left") {
```

Comment:
`opt$logo_right == FALSE` should be `!opt$logo_right`

Reply:
Ok. I mean I prefer this for readability sometimes, but point taken. You might want to do the same in `validate_elt`.

Reply:
OK! It's just perilously close to a bad programming meme, so I think it's good to avoid it. Feel free to update `validate_elt` if you like too, otherwise just do `opt$logo_...`

Reply:
Oh you'll have to share the meme with me! I did update `validate_elt` too.